pkg/big.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 Chair of Applied Cryptography, Technische Universität
+// Darmstadt, Germany. All rights reserved. This file is part of
+// perun-eth-demo. Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package pkg
+
+import "math/big"
+
+// BigInt wraps a golang math/big.Int.
+// All functions on BigInt have their equivalent in the documentation below.
+// See https://golang.org/pkg/math/big/#Int
+type BigInt struct {
+	big *big.Int
+}
+
+// NewBigIntFromBytes creates a BigInt from a byte slice.
+func NewBigIntFromBytes(data []byte) *BigInt {
+	return &BigInt{new(big.Int).SetBytes(data)}
+}
+
+// NewBigIntFromInt64 creates a BigInt from an int64.
+func NewBigIntFromInt64(v int64) *BigInt {
+	return &BigInt{new(big.Int).SetInt64(v)}
+}
+
+// Add wraps math/big/Int.Add
+func (b *BigInt) Add(x, y *BigInt) *BigInt {
```

Comment:
right now this would be quite painful to use, even more so than in Go, because an empty receiver has to be created first. Maybe we improve the API here a bit and change `Add` and `Sub` to methods that internally create the receiver and then return the result, like
```suggestion
func (b *BigInt) Add(y *BigInt) *BigInt {
 	return &BigInt{new(big.Int).Add(b.big, y.big)}
}
```

Reply:
Yes i also felt weird while writing that but was now sure if we should omit the go API.  
I will do it like you proposed now.

Reply:
Wait no, `Add` normally changes the receiver, that should stay AFAIK.